lib/kinesis_client.rb

```diff
@@ -7,42 +7,96 @@
 class KinesisClient
   attr_reader :config, :avro
 
-  def initialize(config)
+  def initialize(config, batch_size = 1, automatically_push = true)
```

Comment:
I like the catch-all `config` param. Is there a reason `stream_name`, `schema_string`, and `profile` are sent in via the `config` hash, but `batch_size` and `automatically_push` are not? I feel like ordered parameters may be appropriate when they're required, but a catch-all hash param suits optional params - particularly as they grow in number. It's easy to make mistakes when the number of ordered parameters grows.

Reply:
Good point. I'll put those in `config`